tweet/index.js

```diff
@@ -22,37 +22,31 @@ module.exports = function (fastify, opts, next) {
     data: opts
   })
 
-  fastify.register(function (fastify, opts, done) {
+  fastify.register(async function (fastify, opts) {
     fastify.register(require('fastify-mongodb'), {
       url: fastify.config.TWEET_MONGO_URL
     })
 
-    fastify.register(fp(function decorateWithTweetCollection (fastify, opts, done) {
+    fastify.register(fp(async function decorateWithTweetCollection (fastify, opts) {
       fastify.decorate('tweetCollection', fastify.mongo.db.collection('users'))
-      done()
     }))
 
-    fastify.register(function (fastify, opts, done) {
-      require('./mongoCollectionSetup')(fastify.mongo.db, fastify.tweetCollection, done)
+    fastify.register(async function (fastify, opts) {
+      require('./mongoCollectionSetup')(fastify.mongo.db, fastify.tweetCollection)
```

Comment:
should this be awaited or returned?

Reply:
I don't understand. If you like more, I can move the function definition on the top of the file keeping the plugin registration cleaner.

I have used `async` to avoid the function call

Reply:
You should likely do:

```js
await require('./mongoCollectionSetup')(fastify.mongo.db, fastify.tweetCollection)
 ```

Reply:
done